velox/core/PlanNode.h

```diff
@@ -1422,14 +1422,27 @@ class MergeJoinNode : public AbstractJoinNode {
   static PlanNodePtr create(const folly::dynamic& obj, void* context);
 };
 
-// Cross join.
-class CrossJoinNode : public PlanNode {
+/// Represents inner/outer nested loop joins. Translates to an
+/// exec::NestedLoopJoinProbe and exec::NestedLoopJoinBuild. A separate pipeline
+/// is produced for the build side when genreating exec::Operators. Nested loop
+/// join supports both euqal and non-equal joins. Expressions specified in
+/// joinCondition are evaluated on every combination of left/right tuple, to
+/// emit result.
+class NestedLoopJoinNode : public PlanNode {
  public:
-  CrossJoinNode(
+  NestedLoopJoinNode(
+      const PlanNodeId& id,
+      JoinType joinType,
+      TypedExprPtr joinCondition,
+      PlanNodePtr left,
+      PlanNodePtr right,
+      const RowTypePtr& outputType);
```

Comment:
pass by value and std::move

Reply:
I find inconsistent usage of `outputType` parameter across `PlanNode.h`:
- const reference is used by `ArrowStreamNode`, `TableScanNode` and `TableWriteNode`
- left value is used by `PartitionedOutputNode`
- const value is used by `AbstractJoinNode` and hence `HashJoinNode`/`MergeJoinNode`
Should we unify them all to pass by value? (take ownership)

Reply:
It would be nice to unify these.

Reply:
Got it, will do.

Reply:
Here is the PR #4683.